spec/features/authentication/set_up_enhanced_two_factor_authentication_spec.rb

```diff
@@ -1,161 +1,311 @@
+        expect(notification_instance).to_not have_received(:send_email)
+      end
     end
-  end
 
-  context "when admin user requests to re-send TOTP email" do
-    before do
-      choose "Email"
-      click_on "Continue"
-      click_on "Complete setup"
-      click_on "we can try sending it again"
+    context "when user chooses email as the 2FA method" do
+      before do
+        choose "Email"
+        click_on "Continue"
+      end
+
+      it "explains that 2FA codes will be sent by email" do
+        expect(page).to have_content("Two-factor authentication")
+        expect(page).to have_content("We can send security codes to the email address")
+        expect(page).to have_button("Complete setup")
+        expect(page).to have_link("Back")
+      end
+
+      it "does not send an email" do
+        expect(notification_instance).to_not have_received(:send_email)
+      end
     end
 
-    it "allows requesting to re-send TOTP email" do
-      expect(page).to have_button("Resend email")
+    context "when user completes email-based 2FA setup" do
+      before do
+        choose "Email"
+        click_on "Continue"
+        click_on "Complete setup"
+      end
+
+      it "confirms that an email with 2FA code has been sent" do
+        expect(page).to have_content("We have emailed you a link to sign in to GovWifi")
+      end
+
+      it "explains what can be done in case email is not received" do
+        expect(page).to have_css("#resend-email")
+      end
+
+      it "sends an email" do
+        expect(notification_instance).to have_received(:send_email)
+      end
+
+      context "the user resends the email" do
+        before :each do
+          click_link "resend-email"
+        end
+
+        it "explains the email can be sent again" do
+          expect(page).to have_content("Emails sometimes take a few minutes to arrive.")
+          expect(page).to have_content("If you do not receive the email, we can send you a new one.")
+        end
+
+        it "does not send further emails" do
+          expect(notification_instance).to have_received(:send_email).once
+        end
+
+        context "the user confirms" do
+          before :each do
+            click_on "Resend email"
+          end
+
+          it "sends another email" do
+            expect(notification_instance).to have_received(:send_email).twice
+          end
+
+          it "explains another email has been sent" do
+            expect(page).to have_content("We have emailed you another link to sign in to GovWifi.")
+          end
+        end
+      end
+
+      context "The user signs out and back in again" do
+        before :each do
+          click_on "Sign out"
+          visit root_path
+
+          fill_in "Email", with: "tom@gov.uk"
+          fill_in "Password", with: correct_password
+          click_on "Continue"
+        end
+
+        it "sends an email again" do
+          expect(notification_instance).to have_received(:send_email).twice
```

Comment:
I'm probably missing something obvious, but why is email being sent twice in this case?

Reply:
The email has already been sent in the previous ```before``` blocks and now the user clicks 'resend' which means it should be sent again.